zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1679,6 +1680,13 @@ States getState() {
         return state;
     }
 
+    /**
+     * Returns true if SSL connection was requested and the actual ClientCnxnSocket supports SSL
+     */
+    public boolean isSSL() {
```

Comment:
I would go for 'isSecure'

But honestly I don't see value in having this method.

How will the application benefit from this API?

Reply:
well... I actually added this after your comment / change request. I thought you would think this to be a good idea :)
> Is there a way to check that we are using a secure connection?

I can imagine a use-case, when the application wants to really verify that the channel is secure before sending any sensitive data to ZooKeeper. You can simply override the SSL related config with a system property when you start the application (even by accident, simply just by forgetting to specify the system property to use Netty socket implementation), and then you run your application assuming you use SSL, but you don't.

But anyway, I don't want to make the API more complex... if you think it is unnecessary, we can drop it and revert my last commit.